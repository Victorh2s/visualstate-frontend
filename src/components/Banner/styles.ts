import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`

  width: 100%;
  height: 22rem;
  overflow: hidden;
  cursor: pointer;


  `}
`;

export const Banner = styled.img`
  ${({ theme }) => css`
  width: 100%;

  `}
`;

export const GladientVertical = styled.div`
  ${({ theme }) => css`


  width: inherit;
  height: inherit;
  background: linear-gradient(to bottom, #0d0c0c 5%, transparent 100%);


  `}
`;

export const DivButtons = styled.div`
  ${({ theme }) => css`

  margin-top: -3rem;
  float: right;
  width: 10%;
  height:3rem;
  border-top-left-radius: 10px;
  background: ${theme.colors.dark};
  text-align: center;
  cursor: pointer;

  > svg {
    color: ${theme.colors.white};
  `}
`;

export const Edit = styled.button`
  ${({ theme }) => css`
  border: none;
  width: 3rem;
  height:3rem;
  background: transparent;
  cursor: pointer;

  > svg {
    color: ${theme.colors.white};
  `}
`;
