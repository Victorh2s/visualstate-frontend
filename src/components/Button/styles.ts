import styled, { css } from 'styled-components';
import { ButtonProps } from '.';

export const Button = styled.button<Pick<ButtonProps, 'color'>>`
  ${({ theme, color }) => css`
    background: ${theme.colors[color]};
    border: none;
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.small};
    padding: ${theme.spacings.small} ${theme.spacings.medium};
    cursor: pointer;
    border-radius: 1rem;
    transition: all 300ms ease-in-out;
    display: flex;
    align-items: center;
    justify-content: center;

    &:focus {
      outline: none;
      box-shadow: 0 0 ${theme.spacings.small} ${theme.colors[color]};
      filter: brightness(110%);
    }

    &:hover {
      filter: brightness(90%);
    }

    &:disabled {
      background: ${theme.colors.gray8};
      color: ${theme.colors.white};
      cursor: not-allowed;

      &:hover {
        filter: none;
      }
    }

    > svg {
      width: 2rem;
      height: 2rem;
      margin-left: 1rem;
    }
  `}
`;
