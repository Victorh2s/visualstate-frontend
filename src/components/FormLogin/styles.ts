import styled, { css } from 'styled-components';

export const Form = styled.form`
`;

export const ButtonWrapper = styled.div``;

export const Container = styled.div`
    text-align: center;
    width: 35vw;
`;

export const Wrapper = styled.div`
  ${({ theme }) => css`
    width: 100%;
    margin-bottom: ${theme.spacings.large};

  `}
`;

export const InputWrapper = styled.div`
  ${({ theme }) => css`
    position: relative;

    > svg {
      position: absolute;
      top: 50%;
      transform: translate(0, -50%);
      right: 1rem;
      width: 2.5rem;
      height: 2.5rem;
      color: ${theme.colors.gray6};
      z-index: ${theme.layers.layer1};

    }

    *:focus ~ svg {
      color: ${theme.colors.primary};

    }

  `}
`;

export const Input = styled.input`
  ${({ theme }) => css`
    border: 1px solid ${theme.colors.gray3};
    width: 100%;
    height: 100%;
    font-size: ${theme.font.sizes.normal};
    transition: ${theme.transitions.fast};
    padding: ${theme.spacings.small} ${theme.spacings.xsmall};
    background: ${theme.colors.white};
    padding-right: 3.7rem;
    border-radius: ${theme.spacings.tiny};
    outline: none;

    &::placeholder {
      visibility: hidden;
      opacity: 0;
    }

    &:focus {
      border: ${theme.spacings.xxtiny} solid ${theme.colors.primary};
      box-shadow: 0 0 ${theme.spacings.xtiny} 0 ${theme.colors.primary};
      background: ${theme.colors.white};
    }

    &:focus
      + ${Label},
      &:not(:placeholder-shown)
      + ${Label},
      &:-webkit-autofill
      + ${Label} {
      top: 0;
      transform: translate(0, -50%);
      font-size: ${theme.font.sizes.xsmall};
      color: ${theme.colors.white};
      background: ${theme.colors.primary};
      filter: none;
    }


  `}
`;

export const Label = styled.label<{ element: string }>`
  ${({ theme, element }) => css`
    display: inline;
    position: absolute;
    top: 50%;
    transform: translate(0, -50%);
    left: ${theme.spacings.xsmall};
    font-size: ${theme.font.sizes.small};
    height: ${theme.font.sizes.small};
    transition: ${theme.transitions.fast};
    background: ${theme.colors.white};
    padding: 0.5rem 1rem;
    display: flex;
    align-items: center;
    line-height: 1;
    z-index: ${theme.layers.layer1};
    color: ${theme.colors.gray6};
    border-radius: ${theme.spacings.tiny};

    ${
      element === 'textarea' &&
      css`
      top: ${theme.spacings.large};
      transform: translate(0, -50%);
    `
    }
  `}
`;

export const DivButton = styled.div`

`;
export const DivLink = styled.div`
  ${({ theme }) => css`

    margin-bottom: ${theme.spacings.large};
    > a{
      text-decoration: none;
    }
  `}
`;

export const button = styled.button`
  ${({ theme }) => css`

    background: ${theme.colors.primary};
    border: none;
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.small};
    padding: ${theme.spacings.small} ${theme.spacings.medium};
    cursor: pointer;
    border-radius: 1rem;
    transition: all 300ms ease-in-out;
    display: flex;
    margin: 0 auto;

    &:focus {
      outline: none;
      box-shadow: 0 0 ${theme.spacings.small} ${theme.colors.primary};
      filter: brightness(110%);
    }

    &:hover {
      filter: brightness(90%);
    }



    > svg {
      width: 2rem;
      height: 2rem;
      margin-left: 1rem;
    }
  `}
`;

export const ErrorMessage = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.warning};
    margin: ${theme.spacings.xxsmall} 0 ${theme.spacings.medium};
  `}
`;
