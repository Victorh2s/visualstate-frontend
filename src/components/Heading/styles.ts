import styled, { css, DefaultTheme } from 'styled-components';
import { HeadingProps } from '.';

const titleSize = {
  supersmall: (theme: DefaultTheme) => css`
  font-size: ${theme.font.sizes.xxsmall};
    ${mediaFont(theme)};
  `,
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.medium};
    ${mediaFont(theme)};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.large};
    ${mediaFont(theme)};
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xlarge};
    ${mediaFont(theme)};
  `,
  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xhuge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme: DefaultTheme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.medium};
  }
  @media ${theme.media.lteSmall} {
    font-size: ${theme.font.sizes.small};
  }
  @media ${theme.media.lteSuperSmall} {
    font-size: ${theme.font.sizes.xxsmall};
  }
`;

const titleCase = (uppercase: boolean) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1<HeadingProps>`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.dark : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
