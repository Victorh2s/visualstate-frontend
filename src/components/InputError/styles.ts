import styled, { css } from 'styled-components';

export const ErrorMessage = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.warning};
    margin: ${theme.spacings.xxsmall} 0 ${theme.spacings.large};
  `}
`;
