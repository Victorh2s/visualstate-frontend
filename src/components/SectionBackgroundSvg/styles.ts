import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
  padding: 0;
  background-image: url('/assets/uploads/details.svg');
  background-size: cover;
  background-position: center center;
    color: ${theme.colors.dark};
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
