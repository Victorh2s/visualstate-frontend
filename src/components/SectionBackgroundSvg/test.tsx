import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { SectionBackgroundSvg, SectionBackgroundSvgProps } from '.';

const props: SectionBackgroundSvgProps = {
  children: 'Texto',
};
describe('<SectionBackgroundSvg />', () => {
  it('should render SectionBackgroundSvg with background false', () => {
    const { container } = renderTheme(<SectionBackgroundSvg {...props} />);

    expect(screen.getByText('Texto')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it('should render SectionBackgroundSvg with background true', () => {
    const { container } = renderTheme(<SectionBackgroundSvg {...props} />);

    expect(screen.getByText('Texto')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
