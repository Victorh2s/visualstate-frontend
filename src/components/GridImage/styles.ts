import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
${({ theme }) => css`
padding: 2rem;
padding-bottom: 10rem;
 > ${TextComponent}{
  text-align: center;
  margin-bottom: ${theme.spacings.xhuge}
}
${HeadingContainer}{
  text-align: center;
}
`}
`;

export const Grid = styled.div`
${({ theme }) => css`
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px, 1fr));
  gap: ${theme.spacings.large}
`}
`;

export const GridElement = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 300ms ease-in-out;
  &:hover{
    transform: scale(1.2);
  }
`;
