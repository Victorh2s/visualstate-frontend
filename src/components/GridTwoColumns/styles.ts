import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export type translatey = {
  translatey: number;
};

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    overflow-x: hidden;
    align-items: center;
    width: 100%;
    gap: ${theme.spacings.large};
    @media ${theme.media.lteOrEqMedium} {
      display: block;
      text-align: center;
      align-items: center;
      overflow-x: hidden;
    }
    @media ${theme.media.lteOrEqSmallest} {
      width: 70%;
  }
    ${Title} {
      margin-bottom: 0;
    }
  `}
`;

export const TextContainer = styled.div`
    width: 35rem;
    display: flex;
    overflow-x: hidden;
    margin: 0 auto;
    padding-bottom: 0.1rem;
`;

export const Ytext = styled.div<translatey>`
  ${({ theme, translatey }) => css`
    display: flex;
    transform: translateX(-${translatey}rem);
    transition: all 400ms ease-in-out;
    padding-bottom: 0.1rem;
    @media ${theme.media.lteOrEqSmallest} {
      transform: translateX(-${translatey - 11}rem);
      transition: all 400ms ease-in-out;
    }
`}
`;

export const HeadingHeight = styled.div`
 ${({ theme }) => css`
  width: 35rem;
  padding-bottom: 0.1rem;
  @media ${theme.media.lteOrEqSmallest} {
      width: 25rem;
  }
  `}
`;

export const Carrousel = styled.div`
 ${({ theme }) => css`
  height: 40rem;
  margin: 0 auto;
  overflow-y: hidden;
  @media ${theme.media.lteOrEqSmallest} {
    height: 25rem;
  }

  `}
`;

export const Yvideo = styled.div<translatey>`
  ${({ theme, translatey }) => css`
    transform: translateY(-${translatey}rem);
    transition: all 400ms ease-in-out;
    @media ${theme.media.lteOrEqMedium} {
      transform: translateY(-${translatey + 12}rem);
      transition: all 400ms ease-in-out;
  }

  `}
`;

export const SourceContainer = styled.div`
  width: 100%;
  height: 50rem;
`;

export const videos = styled.video`
  ${({ theme }) => css`
  width: 45vw;
  height: 50rem;
  @media ${theme.media.lteOrEqMedium} {
    width: 80%;
    height: 50rem;
    align-items: center;
  }
  @media ${theme.media.lteOrEqSmallest} {
    width: 60%;
    height: 50rem;
    align-items: center;
  }
    `}
`;
