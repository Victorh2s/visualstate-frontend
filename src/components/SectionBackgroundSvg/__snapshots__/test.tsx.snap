// Vitest Snapshot v1

exports[`<SectionBackgroundSvg /> > should render SectionBackgroundSvg with background false 1`] = `
.c0 {
  padding: 0;
  background-image: url('/assets/uploads/details.svg');
  background-size: cover;
  background-position: center center;
  color: #07090D;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 0 auto;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Texto
    </div>
  </div>
</div>
`;

exports[`<SectionBackgroundSvg /> > should render SectionBackgroundSvg with background true 1`] = `
.c0 {
  padding: 0;
  background-image: url('/assets/uploads/details.svg');
  background-size: cover;
  background-position: center center;
  color: #07090D;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 0 auto;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Texto
    </div>
  </div>
</div>
`;
